flake.nix

```diff
@@ -13,57 +12,63 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
         inherit system;
         overlays = [(import rust-overlay)];
-      };
+      });
+    gitRev = self.rev or self.dirtyRev or null;
+  in {
+    packages = eachSystem (system: {
+      helix = pkgsFor.${system}.callPackage ./default.nix {inherit gitRev;};
+      /**
+      The default Helix build. Uses the latest stable Rust toolchain, and unstable
+      nixpkgs.
+
+      The build inputs can be overriden with the following:
+
+      packages.${system}.default.override { rustPlatform = newPlatform; };
+
+      Overriding a derivation attribute can be done as well:
 
+      packages.${system}.default.overrideAttrs { buildType = "debug"; };
+      */
+      default = self.packages.${system}.helix;
+    });
+    checks = eachSystem (system: let
       # Get Helix's MSRV toolchain to build with by default.
-      msrvToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
-      msrvPlatform = pkgs.makeRustPlatform {
+      msrvToolchain = pkgsFor.${system}.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      msrvPlatform = pkgsFor.${system}.makeRustPlatform {
         cargo = msrvToolchain;
         rustc = msrvToolchain;
       };
     in {
-      packages = rec {
-        helix = pkgs.callPackage ./default.nix {inherit gitRev;};
-
-        /**
-        The default Helix build. Uses the latest stable Rust toolchain, and unstable
-        nixpkgs.
-        
-        The build inputs can be overriden with the following:
-         
-        packages.${system}.default.override { rustPlatform = newPlatform; };
-         
-        Overriding a derivation attribute can be done as well:
-        
-        packages.${system}.default.overrideAttrs { buildType = "debug"; };
-        */ 
-        default = helix;
-      };
-
-      checks.helix = self.outputs.packages.${system}.helix.override {
+      helix = self.outputs.packages.${system}.helix.override {
         buildType = "debug";
         rustPlatform = msrvPlatform;
       };
+    });
 
-      # Devshell behavior is preserved.
-      devShells.default = let
+    # Devshell behavior is preserved.
+    devShells = eachSystem (system: {
```

Comment:
Another place you can just use `mapAttrs` directly instead of `eachSystem`.

Reply:
For `devShells`, I always `mapAttrs op pkgsFor`, because this is a *consumer* of `pkgs`, not a producer of `packages`. Here, I never waver.